stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1369,6 +1376,12 @@ void TaskGroupBase::resumeWaitingTask(
         // locks) because we know that the child task is completed and
         // we can't be holding its locks ourselves.
         _swift_taskGroup_detachChild(asAbstract(this), completedTask);
+        if (hadErrorResult) {
+          // We only used the task to keep the error in the future fragment around
+          // so now that we emitted the error and detached the task, we are free to release the task immediately.
+          auto error = reinterpret_cast<SwiftError *>(result.storage);
+          swift_release(completedTask); // we need to do this if the error is a class
+        }
```

Comment:
This is somewhat weird; releasing the error does not do the right thing it seems (over releasing);
but releasing the task seems to be right since we retained this task to keep the error around. We could instead try to store only the error but that's a bigger refactoring.